cmd/rm-main.go

```diff
@@ -212,9 +212,9 @@ func checkRmSyntax(ctx context.Context, cliCtx *cli.Context, encKeyDB map[string
 			"You cannot specify --version-id with any of --versions, --rewind and --recursive flags.")
 	}
 
-	if isNoncurrentVersion && !isVersions {
+	if isNoncurrentVersion && !isVersions && !isRecursive {
```

Comment:
```suggestion
	if !(isNoncurrentVersion && isVersions && isRecursive) {
```

Reply:
All false will trigger the error, which isn't intended, I suppose:
```
func main() {
	var isNoncurrentVersion, isVersions, isRecursive bool
	fmt.Println(isNoncurrentVersion && !isVersions && !isRecursive)
	fmt.Println(!(isNoncurrentVersion && isVersions && isRecursive))
}
```

Reply:
yes, it should be if isNoncurrentVersion && !(isVersions && isRecursive) {